### PR TITLE
Add new keyboard "Heavy Left"

### PR DESCRIPTION
**Pull request by takashicompany**
## Description

This is a 95-key split type Japanese keyboard.  
It has a numeric keypad on the left hand side, which can be used not only for entering numbers but also as a dedicated macro pad.  
It also supports key switch replacement with MX sockets and LED underglow.  

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/takashicompany/heavy_left/config.h [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+#define RGB_DI_PIN D3
+#ifdef RGB_DI_PIN
+#    define RGBLED_NUM 6 * 2
+#    define RGBLED_SPLIT {6, 6}
+#    define RGBLIGHT_HUE_STEP 8
+#    define RGBLIGHT_SAT_STEP 8
+#    define RGBLIGHT_VAL_STEP 8
+#    define RGBLIGHT_LIMIT_VAL 255 /* The maximum brightness level */
+#    define RGBLIGHT_SLEEP  /* If defined, the RGB lighting will be switched off when the host goes to sleep */
+#    define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/takashicompany/heavy_left/config.h:74]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/takashicompany/heavy_left/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include "keymap_jp.h"
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+    KC_00
```
Comment (drashna) [inline keyboards/takashicompany/heavy_left/keymaps/default/keymap.c:21]:
```suggestion
    KC_00 = SAFE_RANGE`
```

File: keyboards/takashicompany/heavy_left/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include "keymap_jp.h"
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+    KC_00
```
Comment (drashna) [inline keyboards/takashicompany/heavy_left/keymaps/via/keymap.c:21]:
```suggestion
    KC_00 = USER00,
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/takashicompany/heavy_left/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include "keymap_jp.h"
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+    KC_00 = SAFE_RANGE`
```
Comment (mtei) [inline keyboards/takashicompany/heavy_left/keymaps/default/keymap.c:21]:
```suggestion
    KC_00 = SAFE_RANGE
```

File: keyboards/takashicompany/heavy_left/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    )
+
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case KC_00:
+            if (record->event.pressed) {
+                // when keycode QMKBEST is pressed
```
Comment (fauxpark) [inline keyboards/takashicompany/heavy_left/keymaps/via/keymap.c:64]:
```suggestion
```

File: keyboards/takashicompany/heavy_left/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+                SEND_STRING("00");
+            }
+            break;
```
Comment (fauxpark) [inline keyboards/takashicompany/heavy_left/keymaps/via/keymap.c:67]:
```suggestion
            return false;
```

File: keyboards/takashicompany/heavy_left/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case KC_00:
+            if (record->event.pressed) {
+                // when keycode QMKBEST is pressed
+                SEND_STRING("00");
+            }
+            break;
```
Comment (fauxpark) [inline keyboards/takashicompany/heavy_left/keymaps/default/keymap.c:67]:
```suggestion
            return false;
```

File: keyboards/takashicompany/heavy_left/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+    ),
+    
+    LAYOUT(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    )
+
```
Comment (fauxpark) [inline keyboards/takashicompany/heavy_left/keymaps/default/keymap.c:57]:
```suggestion
```

**Review by mtei — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**